src/Service/NuRaftStateManager.cpp

```diff
@@ -24,22 +24,36 @@ NuRaftStateManager::NuRaftStateManager(int id_, const Poco::Util::AbstractConfig
     cur_cluster_config = parseClusterConfig(config_, "keeper.cluster");
 }
 
+ptr<cluster_config> NuRaftStateManager::get_cluster_config() const
+{
+    std::lock_guard<std::mutex> l(cluster_config_lock_);
+    ptr<cluster_config> ret = cur_cluster_config;
+    return ret;
+}
+
+void NuRaftStateManager::set_cluster_config(const ptr<cluster_config>& new_config)
+{
+    std::lock_guard<std::mutex> l(cluster_config_lock_);
+    cur_cluster_config = new_config;
+    LOG_INFO(log, "set config with log index {}", cur_cluster_config->get_log_idx());
+}
+
 ptr<cluster_config> NuRaftStateManager::load_config()
 {
     if (!Poco::File(cluster_config_file).exists())
     {
         LOG_INFO(log, "load config with initial cluster config.");
-        return cur_cluster_config;
+        return get_cluster_config();
     }
 
     std::unique_ptr<ReadBufferFromFile> read_file_buf = std::make_unique<ReadBufferFromFile>(cluster_config_file, 4096);
     size_t size;
     readVarUInt(size, *read_file_buf);
     ptr<nuraft::buffer> buf = nuraft::buffer::alloc(size);
     read_file_buf->readStrict(reinterpret_cast<char *>(buf->data()), size);
-    cur_cluster_config = nuraft::cluster_config::deserialize(*buf);
-    LOG_INFO(log, "load config with log index {}", cur_cluster_config->get_log_idx());
```

Comment:
Why delete the logging?

Reply:
I just move log into function `setClusterConfig`, maybe it's better to print log after call it.